explanation/performance/perf-epp.md

```diff
@@ -0,0 +1,69 @@
+(perf-epp)=
+
+# Energy Performance Preference
+
+Energy Performance Preference (EPP) is a CPU power-management feature available on many modern processors (particularly Intel and AMD). It allows the operating system or user to influence the CPU's balance between performance and power consumption. Instead of forcing a strict maximum performance or minimum power mode, EPP provides a hint to the processor’s hardware power control logic about the desired balance.
+
+The actual effect depends on the CPU model and firmware, but in general, lower EPP values tell a processor to favor performance while higher values tell it to favor lower power consumption.
+
+## Caveats
+
+* EPP is only a *preference* or hint to the hardware. It does not guarantee that the CPU will behave in a precise or deterministic way.
+
+* EPP value interpretations will often vary between different CPUs.
+
+* Hardware support is required. On some processors, EPP may have no effect even if it appears to be configurable in the OS.
+
+* Some systems may require an update to BIOS settings to allow OS control of power regulation.
+
+## EPP values on Ubuntu
+
+On Ubuntu, the EPP for each core on a CPU is stored in `/sys/devices/system/cpu/cpu#/cpufreq/energy_performance_preference`. To check the current status on cpu0, for example, you can run:
+
+```bash
+cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_preference
+```
+
+This file will normally contain one of the following named profiles:
+
+   * `performance` - Prioritize responsiveness and raw performance.
+   * `balance_performance` - Lean toward performance, with lower power consumption.
+   * `balance_power` - Lean toward reduced power, with some additional performance.
+   * `power` - Prioritize a reduction in power consumption.
+   * `default` - Use the default mix of power and performance as recommended by Ubuntu.
+
+The full list of options is provided in `/sys/devices/system/cpu/cpu#/cpufreq/energy_performance_available_preferences`. To see them, run:
+
+```bash
+cat /sys/devices/system/cpu/cpu2/cpufreq/energy_performance_available_preferences
```

Comment:
Why does this example use cpu2? (previous used 0)

Reply:
What sort of output do you normally get?

Reply:
oops, copy-paste error from my testing, updating to cpu0

Also adding example output